crates/oxc_isolated_declarations/src/function.rs

```diff
@@ -65,6 +113,8 @@ impl<'a> IsolatedDeclarations<'a> {
                 self.ast.copy(&param.pattern)
             };
 
+        pattern.kind = self.copy_binding_pattern_kind_without_assignments(&pattern.kind);
```

Comment:
This might be an unnecessary copy if no default values exist, but I feel that detecting whether a copy is necessary creates extra work in a different way.